format/FormatNumber.ts

```diff
@@ -464,6 +469,16 @@ function buildFormatConfig(v, precision, zeroPad, withCommas, omitFourDigitComma
     return config;
 }
 
+function countDecimalPlaces(number) {
+    const numberStr = number.toString(),
+        decimalIndex = numberStr.indexOf('.');
+
+    if (decimalIndex === -1) return 0;
+
+    const decimalStr = numberStr.slice(decimalIndex + 1);
+    return decimalStr.replace(/0+$/, '').length; // Remove trailing zeros then return count
```

Comment:
I think this could be faster/simpler with `return decimalIndex === -1 ? 0 : numberStr.length - decimalIndex - 1`. (without the slice or regex)

Pls also type this util fn args/return - good to be super careful / tight on typing in these widely used, library-level utils